engine/verification/assigner/engine.go

```diff
@@ -147,14 +149,36 @@ func (e *Engine) processChunk(chunk *flow.Chunk, resultID flow.Identifier) {
 		return
 	}
 
+	e.metrics.OnChunkProcessed()
+
 	// notifies chunk queue consumer of a new chunk
 	e.newChunkListener.Check()
 	log.Debug().Msg("chunk locator successfully pushed to chunks queue")
 }
 
-// ProcessFinalizedBlock indexes the execution receipts included in the block, and handles their chunk assignments.
-// Once it is done handling all the receipts in the block, it notifies the block consumer.
+// ProcessFinalizedBlock is the entry point of assigner engine. It pushes the block down the pipeline with tracing on it enabled.
+// Through the pipeline the execution receipts included in the block are indexed, and their chunk assignments are done, and
+// the assigned chunks are pushed to the chunks queue, which is the output stream of this engine.
+// Once the assigner engine is done handling all the receipts in the block, it notifies the block consumer.
 func (e *Engine) ProcessFinalizedBlock(block *flow.Block) {
+	blockID := block.ID()
+	span, ok := e.tracer.GetSpan(blockID, trace.VERProcessFinalizedBlock)
+	ctx := context.Background()
+	if !ok {
+		span = e.tracer.StartSpan(blockID, trace.VERProcessFinalizedBlock)
```

Comment:
you might need to use StartSpanFromParent here

Reply:
I think we could not use `StartSpanFromParent` here. This is the entry point of a block to our engine, and block consumer guarantees that this engine will see each block _only once_. Hence, there is no prior parent span with this block ID (as the entity ID) that we [start a span from it](https://github.com/onflow/flow-go/blob/master/module/tracer.go#L27-L31). 

There is an exception though: if the engine (or node) fails while processing this block, and hence its processing is not finished upon restart the block consumer pushes the same block again. And in that case, the tracer recovers the previous span (on line 165) to _resume_ processing this block. In this case, we continue on the same span instead of forking a child from it.